### PR TITLE
Add docker image deployment to the travis config

### LINKED ISSUE #14: [Proposal] Publish a docker image
Currently, we need to install and user license-check via curl and piping to sh

```sh
curl https://raw.githubusercontent.com/teamserverless/license-check/master/get.sh | sh
```

for example here, https://github.com/openfaas/faas-netes/blob/0fe88f3a672c45335e644854f44811a1db287d69/Dockerfile#L6-L7

## Expected Behaviour

In a dockerfile I would prefer to use

```docker
COPY  --from=teamserverless/license-check:v1.0 /usr/bin/license-check /usr/bin/license-check
```

This would allow for better layer caching and, I think, simplify Dockerfiles. It would even allow very simple versioning in dockerfile or for someone to run the license-check via `docker run`.  

## Context

I was thinking about this while update the `faas-cli` project

### PR DESCRIPTION
**Pull request by LucasRoesler**
## Description

- Update the docker images to use Go 1.13
- Add windows build
- Add bash scripts for creating arch specific docker builds

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md))

Resolves #14

## How Has This Been Tested?

I have run all of the new bash scripts locally (except for the docker push), specifically

```sh
./build.sh
./make_tags.sh
./make_manifest.sh
```

The resulting x86 image works as expected

```sh
$ docker run --rm -it openfaas/license-check:latest-dev-x86_64 ./license-check -h
Usage of ./license-check:
  -path string
    	path to scan for licensing
  -verbose
    	verbose output
  -version
    	Print version
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+      export DOCKER_CLI_EXPERIMENTAL=enabled
+
+      ./make_tags.sh
+      ./push_images.sh
+      ./make_manifest.sh
+      docker push $DOCKER_NS/$NAME:$TRAVIS_TAG
+
+      echo $QUAY_PASSWORD | docker login -u=$QUAY_USERNAME --password-stdin quay.io;
+
+      DOCKER_NS=quay.io/$DOCKER_NS
```
Comment (alexellis) [inline .travis.yml]: We can drop quay.io, I don't have time to set it up.

File: make_manifest.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+NAME=license-check
+
+docker manifest create $DOCKER_NS/$NAME:$TAG \
+  $DOCKER_NS/$NAME:$TAG-darwin \
+  $DOCKER_NS/$NAME:$TAG-x86_64 \
+  $DOCKER_NS/$NAME:$TAG-armhf \
+  $DOCKER_NS/$NAME:$TAG-arm64 \
+  $DOCKER_NS/$NAME:$TAG-windows
+
+docker manifest annotate $DOCKER_NS/$NAME:$TAG --arch arm $DOCKER_NS/$NAME:$TAG-darwin
```
Comment (alexellis) [inline make_manifest.sh:18]: What about the two additional IBM architectures we have in the repo?

i.e. s390x and ppc64le

Comment (LucasRoesler) [reply make_manifest.sh:18]: What is the arch for that?  same string?

**Comment by alexellis** [comment]
I believe so, we also need to ensure the static binaries are not affected.

File: make_manifest.sh (hunk 2 above)
Comment (alexellis) [reply make_manifest.sh:18]: Yes, I thought I commented, I think it was below.

Comment (LucasRoesler) [reply make_manifest.sh:18]: Just added these

**Comment by alexellis** [comment]
Sorry @LucasRoesler I can't seem to get this to work.

https://travis-ci.com/teamserverless/license-check/jobs/283353905#L262

```
0.00s$ if [ -z $DOCKER_NS ] ; then export DOCKER_NS=openfaas; fi
after_success.2
2640.01s$ if [ ! -z "$TRAVIS_TAG" ] ; then export DOCKER_CLI_EXPERIMENTAL=enabled
265    ./build.sh ./make_tags.sh ./push_images.sh ./make_manifest.sh docker push $DOCKER_NS/$NAME:$TRAVIS_TAG fi
266/home/travis/.travis/functions: eval: line 111: syntax error: unexpected end of file
```

Anything missing / different from the example? https://github.com/openfaas-incubator/of-watchdog/blob/master/.travis.yml

You can test the changes in this PR simply by adding env-vars to your fork.

Screenshot 2020-02-04 at 10 56 56

**Pull request merged**